tests/sdk/v2/test_dsl.py

```diff
@@ -722,3 +722,35 @@ def my_task():
         pass
 
     assert my_task._dependency_imports == expected_imports
+
+
+class TestInvalidResources:
+    @pytest.mark.parametrize(
+        "cpu", ["1001m", "1500m", "1.5", "2001m", "1.0001k", "1500000u"]
+    )
+    def test_invalud_cpu_resources(self, cpu):
+        @sdk.task(resources=sdk.Resources(cpu=cpu))
+        def t():
+            ...
+
+        @sdk.workflow
+        def wf():
+            return t()
+
+        with pytest.raises(InvalidTaskDefinitionError):
+            wf().model
+
+    @pytest.mark.parametrize(
+        "gpu", ["1001m", "1500m", "1.5", "2001m", "1.0001k", "1500000u", "0.15", "0.5"]
+    )
+    def test_invalud_gpu_resources(self, gpu):
```

Comment:
<!-- Suggestions propose improvements to the current subject. It's important to be explicit and clear on what is being suggested and why it is an improvement. Consider using patches and the blocking or non-blocking decorations to further communicate your intent.  -->
**suggestion:** ‏Same here.
Add tests for known valid resources and tests for a whole number decimal like `"1.0"` or `"2.0"`.